### ISSUE TITLE
Check for available SF right before processing

### ISSUE THREAD

**Issue by stefanbschneider** [RealVNF/coordination-simulation#71]
The simulator should check if an SF is available right before processing, when a flow arrives at the target node, ie, here: https://github.com/RealVNF/coordination-simulation/blob/master/src/coordsim/simulation/flowsimulator.py#L178

Currently, it already checks before sending the flow here: https://github.com/RealVNF/coordination-simulation/blob/master/src/coordsim/simulation/flowsimulator.py#L118

Basically, just move the check to the other function. And test that nothing breaks due to that.

**Comment by qarawlus** [comment]
Just checked, the check happens after the flow was forwarded here:

https://github.com/RealVNF/coordination-simulation/blob/master/src/coordsim/simulation/flowsimulator.py#L117 

I agree that the variable naming is a bit confusing, I will fix it. However, since the 'forwarding' here is just calling a timeout that is equal to the path delay, we are simulating that the check is happening after the flow has been forwarded.

**Comment by stefanbschneider** [comment]
I agree, moving from L118 to L178 doesn't change anything in terms of execution. The check is done at the right time, regarding simulation time. Variable naming is also quite fine for me.

For me, it's just a more logical to check if processing is possible *inside* the `process_flow` function, not before calling it. Then we could always call `process_flow` when we want to process a flow locally and the function would take care of all necessary checks regarding available SFs and resources before doing the actual processing. 

But if you have a strong opinion against it and for keeping it as it is, I'm also fine with it. It's just something small I stumbled over when reading the code.

**Comment by stefanbschneider** [comment]
@qarawlus Did you merge this already?

**Comment by qarawlus** [comment]
Yes, just merged.
